### PR TITLE
Release notes update and in-code version increase to 1.0.3.

### PR DESCRIPTION
**Pull request by DellaBitta**
Targets a release for April 28th, 2021.

### CODE REVIEW

File: RELEASE-NOTES.md [hunk 1]
```diff
@@ -1,3 +1,12 @@
+# Google Cloud IoT Core Device SDK for Embedded C version  1.0.2
+#### April 27, 2021
+
+- Updated the POSIX BSP to use open instead of fopen for more consistent error detection, [#100](https://github.com/GoogleCloudPlatform/iot-device-sdk-embedded-c/pull/100) by [@pnfisher}(https://github.com/pnfisher), thank you!
+- Fixed a non null-terminated issue in message.topic with ‘iotc_user_subscription_callback_’, [#111](https://github.com/GoogleCloudPlatform/iot-device-sdk-embedded-c/issues/111) by [@csobrinho](https://github.com/csobrinho), thank you!
+- Added overflow protection to ‘__iotc_calloc()’ calls.
```
Comment (atigyi) [inline RELEASE-NOTES.md:6]: the two above have PR links, this could have PR link as well, but, I leave this up to you

**Review by atigyi — COMMENTED** [review]
PTAL on comments.

File: RELEASE-NOTES.md (hunk 1 above)
Comment (DellaBitta) [reply RELEASE-NOTES.md]: Done.

File: RELEASE-NOTES.md [hunk 2]
```diff
@@ -1,3 +1,12 @@
+# Google Cloud IoT Core Device SDK for Embedded C version  1.0.2
```
[the remark this answers is not in the document]
Comment (DellaBitta) [reply RELEASE-NOTES.md]: woops, thanks!

**Review by atigyi — APPROVED** [review]
👌

**Pull request merged**